examples/reinforcement_learning/dqn_cartpole.py

```diff
@@ -0,0 +1,199 @@
+    loss = mean_clipped_loss(y, target)
+    Q.cleargrads()
+    loss.backward()
+    opt.update()
+
+
+def main():
+
+    parser = argparse.ArgumentParser(description='Chainer example: DQN')
+    parser.add_argument('--env', type=str, default='CartPole-v0',
+                        help='Name of the OpenAI Gym environment to play')
+    parser.add_argument('--batch-size', '-b', type=int, default=64,
+                        help='Number of transitions in each mini-batch')
+    parser.add_argument('--episodes', '-e', type=int, default=1000,
+                        help='Number of episodes to run')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='dqn_result',
+                        help='Directory to output the result')
+    parser.add_argument('--unit', '-u', type=int, default=100,
+                        help='Number of units')
+    parser.add_argument('--target-type', type=str, default='dqn',
+                        help='Target type', choices=['dqn', 'double_dqn'])
+    parser.add_argument('--reward-scale', type=float, default=1e-2,
+                        help='Reward scale factor')
+    parser.add_argument('--replay-start-size', type=int, default=500,
+                        help=('Number of iterations after which replay is '
+                              'started'))
+    parser.add_argument('--iterations-to-decay-epsilon', type=int, default=5000,
+                        help='Number of steps used to linearly decay epsilon')
+    parser.add_argument('--min-epsilon', type=float, default=0.01,
+                        help='Minimum value of epsilon')
+    parser.add_argument('--target-update-freq', type=int, default=100,
+                        help='Frequency of target network update')
+    parser.add_argument('--record', action='store_true', default=True,
+                        help='Record performance')
+    parser.add_argument('--no-record', action='store_false', dest='record')
+    args = parser.parse_args()
+
+    # Initialize an environment
+    env = gym.make(args.env)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    assert isinstance(env.action_space, gym.spaces.Discrete)
+    obs_size = env.observation_space.low.size
+    n_actions = env.action_space.n
+    if args.record:
+        env = gym.wrappers.Monitor(env, args.out, force=True)
+    reward_threshold = env.spec.reward_threshold
+    if reward_threshold is not None:
+        print('{} defines "solving" as getting average reward of {} over 100 '
+              'consecutive trials.'.format(args.env, reward_threshold))
+    else:
+        print('{} is an unsolved environment, which means it does not have a '
+              'specified reward threshold at which it\'s considered '
+              'solved.'.format(args.env))
+
+    # Initialize variables
+    D = collections.deque(maxlen=10 ** 6)
```

Comment:
It might be nice if there is one line comment that explains this variable. (D is a replay buffer?)